unix-domain-socket/src/main/scala/akka/stream/alpakka/unixdomainsocket/UnixSocketAddress.scala

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.unixdomainsocket
+import java.net.SocketAddress
+import java.nio.file.Path
+
+/**
+ * Represents a path to a file on a file system that a Unix Domain Socket can
+ * bind or connect to.
+ */
+final case class UnixSocketAddress(path: Path) extends SocketAddress
```

Comment:
As this is part of the public API, it should not be a case class, should have a private constructor, and should have `apply` and `create` factory methods in its companion object.